hazelcast/src/main/java/com/hazelcast/map/impl/AbstractEvictableRecordStore.java

```diff
@@ -382,6 +383,16 @@ protected void accessRecord(Record record, long now) {
         setExpirationTime(record, maxIdleMillis);
     }
 
+    protected void mergeRecordExpiration(Record record, EntryView mergingEntry) {
+        if (mergingEntry.getTtl() > 0 && mergingEntry.getTtl() != Long.MAX_VALUE) {
```

Comment:
As you say, those values represent eternal entries, however when an entry is created on cluster ```A``` without a TTL using ```IMap.put(k,v)```, the TTL is received by cluster ```B``` as ```Long.MAX_VALUE``` rather than ```zero```.

Currently the TTL on the record is not being set and the replicated entries do live eternally (as this is the bug I am seeking to address), but if we to pass through ```Long.MAX_VALUE``` this would inappropriately enable expiration scanning on the ```DistributedObject``` as a ```TTL > 0``` is passed to ```markRecordStoreExpirable(long ttl)```.

However if preferred, we could simply set the TTL from the merging entry here, and update ```AbstractEvictableRecordStore.markRecordStoreExpirable(long ttl)``` like so...

```
    protected void markRecordStoreExpirable(long ttl) {
        if (ttl > 0L && ttl < Long.MAX_VALUE) {
            expirable = true;
        }
    }
```

In writing this, that sounds like a more elegant solution, and will update the request as such :smile: